tools/generate_hotkeys.py

```diff
@@ -13,34 +17,135 @@
+    """
+    To compare temp_hotkeys and hotkeys file.
+    """
+    if filecmp.cmp(ORIGINAL_OUTPUT_FILE, TEMP_OUTPUT_FILE, shallow=False):
+        print("Hot Keys already in sync with config/keys.py")
+    else:
+        print("Hot Keys not in sync with config/keys.py")
+        if not generate_flag:
+            exit(1)
+
+
+def generate_hotkeys_file(generate_temp_flag: bool = False) -> None:
+    """
+    Generate check_hotkeys.md in docs folder to check with original hotkeys.md
+    """
+    categories = read_help_categories()
+    with open(TEMP_OUTPUT_FILE, "w") as mdFile:
         mdFile.write(
-            f"## {HELP_CATEGORIES[action]}\n"
-            "|Command|Key Combination|\n"
-            "| :--- | :---: |\n"
+            f"<!--- Generated automatically by tools/{SCRIPT_NAME} -->\n"
+            "<!--- Do not modify -->\n\n# Hot Keys\n"
         )
+        for action in categories.keys():
+            mdFile.write(
+                f"## {HELP_CATEGORIES[action]}\n"
+                "|Command|Key Combination|\n"
+                "| :--- | :---: |\n"
+            )
+            for help_text, key_combinations_list in categories[action].items():
+                various_key_combinations = various_key_combination(
+                    key_combinations_list
+                )
+                mdFile.write(f"|{help_text}|{various_key_combinations}|\n")
+            mdFile.write("\n")
+
+    if not generate_temp_flag:
+        compare_temp_hotkeys_file(True)
+        with open(TEMP_OUTPUT_FILE, "r") as mdFile, open(
+            ORIGINAL_OUTPUT_FILE, "w"
+        ) as original_mdFile:
+            for line in mdFile:
+                original_mdFile.write(line)
+        TEMP_OUTPUT_FILE.unlink()
+        print(f"Hot Keys list saved in {ORIGINAL_OUTPUT_FILE}")
+
+
+def lint_hotkeys_file() -> None:
+    """
+    First compare temp_hotkeys and hotkeys file, if in sync then lint for key description and key duplication
+    """
+    generate_hotkeys_file(True)
+    compare_temp_hotkeys_file()
+    TEMP_OUTPUT_FILE.unlink()
+    # To lint keys description and duplicate keys
+    error_flag = 0
+    categories = read_help_categories()
+    for action in categories.keys():
+        check_duplicate_keys_list: List[str] = []
         for help_text, key_combinations_list in categories[action].items():
-            various_key_combinations = " / ".join(
-                [
-                    " + ".join([f"<kbd>{key}</kbd>" for key in key_combination.split()])
-                    for key_combination in key_combinations_list
-                ]
+            check_duplicate_keys_list.extend(key_combinations_list)
+            check_duplicate_keys_list = [
+                key for key in check_duplicate_keys_list if key not in KEYS_TO_EXCLUDE
+            ]
+            various_key_combinations = various_key_combination(key_combinations_list)
+            # Check description style
+            if not re.match(r"^[a-zA-Z\s/()',&@#:_-]*$", help_text):
```

Comment:
I'm not aware of the style here. Is it basically everything except numbers and full stop?
(as the above doesn't have all special characters?)

Reply:
You can also use `re.compile` once before the for loop to speed up the regex matching.

Reply:
The question above was to see if a regex like `"^[^\d.]*$"` would be more suitable.
I'll wait for @neiljp to clarify the description style here.

Reply:
For the ``help_text`` description style, text string except ``.`` and some ``special characters`` are working for now. But, exactly what ``special_characters`` should be allowed that needs some discussion.

Reply:
I added a separate comment for this just now; the original aim was eg. detect `.`, but something like the original regex above would benefit consistency.